src/index.ts

```diff
@@ -6,24 +6,14 @@ import { Transformer } from "./Transformer";
 
 const tsx = /\.tsx?$/;
 
-// Parse a TypeScript module into an ESTree program.
-function transformFile(input: ts.SourceFile): SourceDescription {
-  const transformer = new Transformer(input);
-  const { ast, fixups } = transformer.transform();
-
-  // NOTE(swatinem):
-  // hm, typescript generates `export default` without a declare,
-  // but rollup moves the `export default` to a different place, which leaves
-  // the function declaration without a `declare`.
-  // Well luckily both words have the same length, haha :-D
-  let code = input.getText();
-  code = code.replace(/(export\s+)default(\s+(function|class))/m, "$1declare$2");
-  for (const fixup of fixups) {
-    code = code.slice(0, fixup.range.start) + fixup.identifier + code.slice(fixup.range.end);
-  }
+const toArray = <T>(value: T): T extends ReadonlyArray<any> ? T : [T] =>
+  Array.isArray(value) ? (value as any) : [value];
 
-  return { code, ast };
-}
+type PragmaMap = Map<"reference", Pragma | Pragma[]>;
+type Pragma = {
```

Comment:
also related to linting: My personal preference is to use `interface` declarations